tests/test_data_processing.py

```diff
@@ -26,14 +38,512 @@
+
+    df_conversion = filter_df(df, "type", ["conversion"])
+    assert len(df_conversion["type"]) == 10
+
+    with pytest.raises(KeyError):
+        filter_df(df, "something", ["conversion"])
+
+    # 2. Test
+    path_file_timeseries = os.path.join(
+        os.path.abspath(os.path.join(this_path, os.pardir)),
+        "_files",
+        "test_stacked.csv",
+    )
+
+    # Read stacked time series
+    df = load_timeseries(path_file_timeseries)
+
+    df_BE_BB = filter_df(df, "region", ["BE_BB"])
+
+    assert (
+        df_BE_BB["region"].values[0] == "BE_BB"
+        and df_BE_BB["region"].values[1] == "BE_BB"
+    )
+    assert len(df_BE_BB["region"]) == 2
+
+    df_var_name = filter_df(
+        df,
+        "var_name",
+        [
+            "flow from BB-biomass-st to BB-electricity",
+            "flow from BB-electricity-liion-battery to BB-electricity",
+        ],
+    )
+
+    assert (
+        df_var_name["var_name"].values[0] == "flow from BB-biomass-st to BB-electricity"
+        and df_var_name["var_name"].values[1]
+        == "flow from BB-electricity-liion-battery to BB-electricity"
+    )
+
+
+def test_df_agg():
+    """
+    This test checks whether
+    1. scalars and
+    2. time series
+    are aggregated by a key
+
+    """
+    # 1. Test
+    path_file_scalars = os.path.join(
+        os.path.abspath(os.path.join(this_path, os.pardir)),
+        "_files",
+        "test_scalars.csv",
+    )
+
+    # Read scalars
+    df = load_scalars(path_file_scalars)
+
+    # Aggregate by region
+    df_aggregated_by_region = df_agg(df, "region")
+
+    # Add expected results of aggregation
+    df_aggregated_expected = pd.DataFrame(
+        data={
+            "id_scal": [None] * 8,
+            "scenario": ["base"] * 8,
+            "name": ["Aggregated by region"] * 8,
+            "var_name": [
+                "capacity",
+                "flow_biomass",
+                "flow_electricity",
+                "costs",
+                "capacity",
+                "flow_biomass",
+                "flow_electricity",
+                "costs",
+            ],
+            "carrier": ["All"] * 8,
+            "region": ["BE", "BE", "BE", "BE", "BB", "BB", "BB", "BB"],
+            "tech": ["All"] * 8,
+            "type": ["All"] * 8,
+            "var_value": [
+                0,
+                0,
+                13470154039.4199,
+                0,
+                400000,
+                2692790078.846871,
+                25900023216.653,
+                108971828.939188,
+            ],
+            "var_unit": ["-"] * 8,
+            "reference": [None] * 8,
+            "comment": [None] * 8,
+        }
+    )
+
+    # Check if results of aggregation equal the expected ones
+    pd.testing.assert_frame_equal(df_aggregated_by_region, df_aggregated_expected)
+
+    # Aggregate by carrier
+    df_aggregated_by_carrier = df_agg(df, "carrier")
+
+    assert list(df_aggregated_by_carrier["var_value"].values) == [
```

Comment:
Here as well, it would be more illustrative and easier to read if the expected file would reside in _files

Reply:
Done with commit https://github.com/rl-institut/oemof-B3/pull/57/commits/10429062c7180571172a6d83d4628f3da593b5ba.